app/javascript/crayons/MultiInput/MultiInput.jsx

```diff
@@ -17,90 +17,219 @@ const ALLOWED_CHARS_REGEX = /([a-zA-Z0-9@.])/;
+    setItems(newArr);
+
+    // We also update the hidden selected items list, so removals are announced to screen reader users
+    selectedItemsRef.current.querySelectorAll('li').forEach((selectionNode) => {
+      if (selectionNode.innerText === clickedItem) {
+        selectionNode.remove();
+      }
+    });
+  };
+
+  // If there is a previous selection, then pop it into edit mode
+  const editPreviousSelectionIfExists = () => {
+    if (items.length > 0 && inputPosition !== 0) {
+      const nextEditIndex =
+        inputPosition !== null ? inputPosition - 1 : items.length - 1;
+
+      const item = items[nextEditIndex];
+      deselectItem(item);
+      enterEditState(item, nextEditIndex);
+    }
+  };
+
+  const enterEditState = (editItem, editItemIndex) => {
+    inputSizerRef.current.innerText = editItem;
+    deselectItem(editItem);
+    setEditValue(editItem);
+    setInputPosition(editItemIndex);
+  };
+
+  const exitEditState = ({ nextInputValue = '' }) => {
+    // Reset 'edit mode' input resizing
+    inputRef.current?.style?.removeProperty('width');
+
+    inputSizerRef.current.innerText = nextInputValue;
+    setEditValue(nextInputValue);
+    setInputPosition(nextInputValue === '' ? null : inputPosition + 1);
+    // Blurring away while clearing the input
+    if (nextInputValue === '') {
+      inputRef.current.value = '';
+    }
+  };
+
+  const allSelectedItemElements = items.map((item, index) => {
+    // When we are in "edit mode" we visually display the input between the other selections
+    // If the item being edited appears before the item being rendered then we set its position to
+    // the index + 1 which matches the order, however, any items that appear after the item that is
+    // being edited will need to increment their position by one to make place for the item being edited.
+
+    // at this point the position is already set
+    const defaultPosition = index + 1;
+    const appearsBeforeInput = inputPosition === null || index < inputPosition;
+    const position = appearsBeforeInput ? defaultPosition : defaultPosition + 1;
+    return (
+      <li
+        key={index}
+        className="c-input--multi__selection-list-item w-max"
+        style={{ order: position }}
+      >
+        <SelectionTemplate
+          name={item}
+          className="c-input--multi__selected"
+          onEdit={() => enterEditState(item, index)}
+          onDeselect={() => deselectItem(item)}
+        />
+      </li>
+    );
+  });
 
   return (
     <Fragment>
+      <span
+        ref={inputSizerRef}
+        aria-hidden="true"
+        className="absolute pointer-events-none opacity-0 p-2"
+      />
+
+      {/* A visually hidden list provides confirmation messages to screen reader users as an item is selected or removed */}
+      <div className="">
+        <p>Selected items:</p>
+        <ul
+          ref={selectedItemsRef}
+          className=" list-none"
```

Comment:
```suggestion
          className=" list-none screen-reader-only"
```
(not blocking) - not sure if we want to make this "hidden from view" here, or in a later PR